common/src/main/kotlin/com/gitlab/kordlib/common/entity/Permission.kt

```diff
@@ -2,51 +2,99 @@ package com.gitlab.kordlib.common.entity
 
 import kotlinx.serialization.KSerializer
 import kotlinx.serialization.Serializable
-import kotlinx.serialization.Serializer
 import kotlinx.serialization.descriptors.PrimitiveKind
 import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
 import kotlinx.serialization.descriptors.SerialDescriptor
 import kotlinx.serialization.encoding.Decoder
 import kotlinx.serialization.encoding.Encoder
-import kotlin.contracts.ExperimentalContracts
-import kotlin.contracts.InvocationKind
-import kotlin.contracts.contract
 
 @Serializable(with = Permissions.Companion::class)
-class Permissions constructor(val code: Int) {
+class Permissions constructor(val code: Int = 0) {
 
     val values: Set<Permission> get() = Permission.values().filter { it.code and code == it.code }.toSet()
 
+    /**
+     * Returns a new [Permissions] that adds [permission] to its [code]
+     */
+
     operator fun plus(permission: Permission): Permissions = Permissions(this.code or permission.code)
 
+    /**
+     * Returns a new [Permissions] that adds [permission] to its [code]
+     */
+
+    operator fun plus(permission: Permissions): Permissions = Permissions(this.code or permission.code)
+
+    /**
+     * Returns a new [Permissions] that removes [permission] from its [code]
+     */
 
     operator fun minus(permission: Permission): Permissions = Permissions(code xor (code and permission.code))
 
+    /**
+     * Returns a new [Permissions] that removes [permission] from its [code]
+     */
+
+    operator fun minus(permission: Permissions): Permissions = Permissions(code xor (code and permission.code))
+
+    /**
+     *  returns a new [Permissions] that removes [Permission] from this [Permissions]'s [code]
+     */
+
+    operator fun Permission.unaryMinus() = minus(this)
+
+    /**
+     *  returns a new [Permissions] that removes [Permissions] from the [code] of an [Permissions]
+     */
+
+    operator fun Permissions.unaryMinus() = minus(this)
+
+    /**
+     *  returns  a new [Permissions] that adds [Permissions] to the [code] of an [Permissions]
+     */
+
+    operator fun Permissions.unaryPlus() = plus(this)
+
+
+    /**
+     *  returns  a new [Permissions] that adds [Permission] to the [code] of an [Permissions]
+     */
+
+    operator fun Permission.unaryPlus() = plus(this)
+
+
+    /**
+     *  checks if a given set of [Permissions] is inside another [Permissions]
+     */
 
     operator fun contains(permission: Permission): Boolean {
         return this.code and permission.code == permission.code
     }
 
-    @OptIn(ExperimentalContracts::class)
-    inline fun copy(block: PermissionsBuilder.() -> Unit): Permissions {
-        contract {
-            callsInPlace(block, InvocationKind.EXACTLY_ONCE)
-        }
-        val builder = PermissionsBuilder(code)
-        builder.apply(block)
-        return builder.permissions()
```

Comment:
Why did we end up removing this? I thought it was quite useful for scoping more complex operations.